compare_template_text/normalize_template_text.py

```diff
@@ -53,13 +55,44 @@ def remove_bulletted_text(self):
         
         self.template_string = re.sub('(?<=\<\<var;name\=\*bullet\*).*?(?=\>\>)','',
                                    self.template_string)
+        self.template_string = re.sub('\<\<var;name=\*bullet.*?\>\>','',self.template_string)
+        self.remove_repeating_chars()
+        return
+    
+    def remove_replaceable_text(self):
+        """The Replaceable Text is found between the <<var copyright tags. This method removes all
+        the text between the original fields from the template and the text. """
+        
+        lists = re.findall('(?<=\<\<var;name=\*;original=).*?(?=;match=.*?\>\>)',
+                           self.template_string)
+        print(lists)
+        for x in lists:
+            if(x.startswith('*')):
+                x = x[1:]
+
+            if(x.endswith('*')):
+                x = x[:-1]
+
+            self.text_string = self.text_string.replace(x,'')
+            self.template_string = self.template_string.replace(x,'')
+            self.template_string = re.sub('\<\<var;name=.*?;original.*?;match.*?\>\>','',self.template_string)
+            self.remove_repeating_chars()
```

Comment:
The match used in the pattern is against the original text.  The match should be against the regex in the `match=`.  To do this, however, you will need to include the text prior and after otherwise it will match too many character.  Most templates match a simple `.*`.

For example, if you have a template:

```
Copyright (C) <<var;name="copyright";original="19yy <name of author>";match=".+">> This program is free software
```
The match should pattern should be `Copyright \(C\) (.+) This program is free software` and the text removed should be the middle 2nd group represented by the `(.+)`.

This algorithm will require capturing the before and after text which is a similar problem to the optional text issue.  You may want to solve both at the same time.

Reply:
@goneall I didnt really get the match regex thing. This regex inside (match = ) just matches the text in the original. So If I remove the original text from both the text and the template, there should be no problem. I tried the above approach in a couple of licenses and there was no mismatch found where there should be a match.

Reply:
See the template format at the end of [Appendix II](https://spdx.github.io/spdx-spec/appendix-II-license-matching-guidelines-and-templates/) of the spec for a description of the var and match= semantics.

Removing the text is essentially treating the var text as optional which will work in some, but not all cases.  The intention of the var text is that within the bounds of the variable text, anything matching the regex in the match= is valid.

If it still isn't clear after reading the spec, let's setup some time to discuss real-time on gitter or I can setup a zoom call.